pkg/downloader/chart_downloader.go

```
@@ -246,22 +246,31 @@ func (c *ChartDownloader) DownloadToCache(ref, version string) (string, *provena
 	if len(digest) == 0 || err != nil {
 		slog.Debug("attempting to download chart", "ref", ref, "version", version)
 		if err != nil && !os.IsNotExist(err) {
+			slog.Error("failed to access chart cache", "error", err, "ref", ref, "version", version)
 			return "", nil, err
 		}
 
 		// Get file not in the cache
 		data, gerr := g.Get(u.String(), c.Options...)
 		if gerr != nil {
+			slog.Error("failed to download chart from remote", "error", gerr, "url", u.String(), "ref", ref, "version", version)
 			return "", nil, gerr
 		}
 
 		// Generate the digest
 		if len(digest) == 0 {
-			digest32 = sha256.Sum256(data.Bytes())
+			// Defensive: check data.Bytes() is not nil/empty
+			bytes := data.Bytes()
+			if len(bytes) == 0 {
```

Does something else check for empty chart data or is it supposed to be valid?

The sh256 should work fine for ""

---

Thanks @TerryHowe  for reviewing.  I checked the codebase and dint find any explicit checks for empty chart data right after downloading. There are other validations for chart contents like metadata and such.

You have a good point about sha256.Sum256 and it does work with empty data. I added this check as in this context, empty chart download likely means somehting went wrong like network issue or bad remote response. This check will make it fail fast and prevents unnecessary processing